framework/widget/src/main/java/org/apache/ofbiz/widget/renderer/macro/MacroFormRenderer.java

```diff
@@ -3005,6 +3006,9 @@ public void makeHyperlinkString(Appendable writer, String linkStyle, String targ
             if (UtilValidate.isNotEmpty(request.getAttribute("id"))) {
                 id = request.getAttribute("id").toString();
             }
+            if (UtilValidate.isNotEmpty(request.getAttribute("title"))) {
+                imgTitle = request.getAttribute("title").toString();
```

Comment:
Given the number of conditional assignments to the imgTitle variable in this method, have you been able to infer any precedence rules for how the title attribute should be determined? 

If so it would be ideal to add tests to demonstrate those rules in MacroFormRendererTest. This would help with the ongoing refactoring effort of MacroFormRenderer to RenderableFtlFormElementsBuilder.

Reply:
@danwatford I came up with this logic : 

- do not mix `image-title` (which can only be rendered as the img title) and `title` (which can only be rendered as the link title)
- if the link's text content is truncated (`description` longer than the `size` attribute), then use the full `description` as link title (we need to give the user the opportunity to access to the the full text content)
- otherwise, use the provided `title` as link's title

I think it's easier to understand this way, what do you think ?

Reply:
The logic sounds good, and even better is that you have asserted it with tests.
Sorry to add to your work, but could you see if you can create the corresponding tests in RenderableFtlFormElementsBuilderTest.

Reply:
I'm no longer requesting that tests are added to RenderableFtlFormElementsBuilderTest.